### DISCUSSION TITLE
Best way to analyze rectified EMG signal

### DISCUSSION THREAD

**Discussion by bnherrerac** [neuropsychology/NeuroKit#725] [Q&A] [answered]
Hello! First of all, thank you all for maintaining this incredible library.

I'm currently working on analyzing EMG signals for injury pattern detection. I've been struggling with analyzing the output EMG signal, because the signal is rectified before being output from the sensor. This plot shows biceps femoris EMG data in green and two angles in blue and orange.

![image](https://user-images.githubusercontent.com/89870545/194900677-026d33a1-1f67-4e95-ba8f-1cabd1d2bab9.png)

I already tried a brief FFT and low-pass filtering code of the EMG signal, but it isn't going very well. This is the frequency plot of the signal:

![image](https://user-images.githubusercontent.com/89870545/194901913-7eaa5824-ace7-4f0f-b7cf-e96df984d837.png)

And this is the result after applying a low-pass filter and reverting back to the time domain:

![image](https://user-images.githubusercontent.com/89870545/194902016-df612d78-ff8a-4a2a-b6f0-a0f1095415b1.png)

I tried using the eda_process function because of how similar they look, but I don't reckon it's the right way. Are there any other NeuroKit functions or pre-processing steps that I could use to gather more information about the signal? Is there a standard procedure for analyzing rectified signals?

Thank you very much!

**Comment by DominiqueMakowski** [comment, marked as answer]
There are a few functions for EMG (https://neuropsychology.github.io/NeuroKit/functions/emg.html), but it depends what do you want to extract. From your signal it seems like there are clear periods of activity / no-activity, so finding the changepoints and locate the "phases" seems like a good start. 

@danibene note to ourselves: it might be good to add a minimal [example](https://neuropsychology.github.io/NeuroKit/examples/index.html#) on the docs about EMG just to mention the EMG-related functions

**Reply by DominiqueMakowski** [reply]
Speaking of which, this repo has been recently published (https://github.com/aalhossary/pyemgpipeline), perhaps we could give it a look and see if there are any features worth adding to NK too
